modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx

```diff
@@ -621,3 +663,346 @@ we've made a few updates to the component in v5:
+    closePopup();
+    console.log('Delete');
+  };
+
+  return (
+    <>
+      <DeleteButton {...targetProps}>Delete Item</DeleteButton>
+      <Popper placement={'bottom'} {...popperProps}>
+        <Popup
+          width={400}
+          heading={'Delete Item'}
+          padding={Popup.Padding.s}
+          handleClose={closePopup}
+        >
+          <p>Are you sure you'd like to delete the item titled 'My Item'?</p>
+
+          <DeleteButton onClick={onDeleteClick}>Delete</DeleteButton>
+          <Button onClick={closePopup}>Cancel</Button>
+        </Popup>
+      </Popper>
+    </>
+  );
+};
+```
+
+**After**
+
+```tsx
+import React from 'react';
+
+import {DeleteButton} from '@workday/canvas-kit-react/button';
+import {
+  Popup,
+  usePopupModel,
+  useCloseOnEscape,
+  useCloseOnOutsideClick,
+  useInitialFocus,
+  useReturnFocus,
+} from '@workday/canvas-kit-react/popup';
+
+export const MyPopup = () => {
+  const model = usePopupModel();
+
+  useCloseOnOutsideClick(model);
+  useCloseOnEscape(model);
+  useInitialFocus(model); // new
+  useReturnFocus(model); // new
+
+  const onDeleteClick = () => {
+    console.log('Delete');
+  };
+
+  return (
+    <Popup model={model}>
+      <Popup.Target as={DeleteButton}>Delete Item</Popup.Target>
+      <Popup.Popper placement={'bottom'}>
+        <Popup.Card width={400} padding="s">
+          <Popup.CloseIcon aria-label="Close" />
+          <Popup.Heading>Delete Item</Popup.Heading>
+          <Popup.Body>
+            <p>Are you sure you'd like to delete the item titled 'My Item'?</p>
+
+            <Popup.CloseButton as={DeleteButton} onClick={onDeleteClick}>
+              Delete
+            </Popup.CloseButton>
+            <Popup.CloseButton>Cancel</Popup.CloseButton>
+          </Popup.Body>
+        </Popup.Card>
+      </Popup.Popper>
+    </Popup>
+  );
+};
+```
+
+The most notable changes is `Popup` is now a container component that takes a `PopupModel` and has
+several subcomponents like `Popup.Target` and `Popup.CloseButton`. These components are hooked up to
+the `PopupModel` via React context and have access to state and events. `Popup.Card` is what the v4
+`Popup` once was.
+
+All behavior hooks, like `useCloseOnEscape` now take a `model` instead of variable parameters. This
+allowed us to fix some subtle bugs. Using the `PopupModel` means all hooks have access to all Popup
+state and events without passing in many parameters.
+
+#### usePopup
+
+In the above example, `usePopup` has been removed. `usePopupModel` should be used instead. All
+subcomponents have an associated behavior hook. For example, `Popup.Target` uses a hook called
+`usePopupTarget`. If you need to use your own components for any reason, these hooks are available.
+`Popup.Target` and `Popup.CloseButton` do not include any styling. They both use `SecondaryButton`
+as the default component rendered. You can change this via the `as` prop. For example, the following
+will render an unstyled button:
```

Comment:
```suggestion
As shown in the example above, `usePopupModel` should now be used instead of `usePopup`. All
subcomponents have an associated behavior hook. For example, `Popup.Target` uses a hook called
`usePopupTarget`. If you need to use your own components for any reason, these hooks are available.
`Popup.Target` and `Popup.CloseButton` do not include any styling. They both render
`SecondaryButton` by default. You can change this via the `as` prop. For example, the following will
render an unstyled button:
```